### PR TITLE
Add a performance metric to measure typing within containers

### PR DESCRIPTION
**Pull request by youknowriad**
Related #46527
 
## What and wy?
This PR adds a performance metric that serves to measuring typing with containers as typing at the root level can remain stable while this degrades (see #46527)

To test you can try running `npm run test:performance packages/e2e-tests/specs/performance/post-editor.test.js` locally or you can take a look at the performance job on this PR.

### CODE REVIEW

**Comment by youknowriad** [comment]
This PR right now includes the fix from https://github.com/WordPress/gutenberg/pull/46527 so the results are very visible for this new metric compared to trunk.

Screenshot 2022-12-14 at 2 23 40 PM

**Comment by youknowriad** [comment]
The fix is now in trunk, so I've rebased this and now the metric should be stable compared to trunk.

File: packages/e2e-tests/specs/performance/post-editor.test.js [hunk 1]
```diff
@@ -151,6 +154,58 @@ describe( 'Post Editor Performance', () => {
 		}
 	} );
 
+	it( 'Typing within containers', async () => {
+		// Measuring block selection performance.
+		await createNewPost();
+		await loadHtmlIntoTheBlockEditor(
```
Comment (tellthemachines) [inline packages/e2e-tests/specs/performance/post-editor.test.js:160]: Would it make sense to instead add this test in another `describe`, so we don't have to go through the setup steps twice? I can't think offhand of any other container-specific tests we might want to run but it's possible there may be others in the future.

**Review by tellthemachines — APPROVED** [review]
Thanks for adding this test! I left a question on the structure below, but I'm happy either way so feel free to ignore it 😄

File: packages/e2e-tests/specs/performance/post-editor.test.js (hunk 1 above)
Comment (youknowriad) [reply packages/e2e-tests/specs/performance/post-editor.test.js:160]: I thought about this a bit and I'm not entirely sure what's the best solution here. Ideally each `it` is independent. Right now, some of them are:

 - typing in container, selecting blocks as both create new posts in the beginning
 - the remaining ones are not independent in the sense that they use whatever is loaded before the test starts (and it seems to be different across metrics depending on where they are placed).

So I do think there's something to be done to improve how things are organized and make it consistent but let's leave this for its dedicated PR and address it for all the tests and not just this particular one.

**Pull request merged**